src/group/libp2p_group_gossip_server.erl

```diff
@@ -99,14 +99,27 @@ handle_call(Msg, _From, State) ->
     {reply, ok, State}.
 
 
-handle_cast({handle_data, Key, Msg}, State=#state{}) ->
+handle_cast({handle_data, Key, Msg, StreamPid}, State=#state{}) ->
     %% Incoming message from a gossip stream for a given key
     case maps:find(Key, State#state.handlers) of
         error -> {noreply, State};
         {ok, {M, S}} ->
             %% Catch the callback response. This avoids a crash in the
             %% handler taking down the gossip_server itself.
-            catch M:handle_gossip_data(Msg, S),
+            try M:handle_gossip_data(Msg, S) of
+                {reply, Reply} ->
```

Comment:
Why not allow the Key for the reply to be set too?

Reply:
"don't cross the streams"